doc/SONiC_OC_NTP_HLD.md

```diff
@@ -0,0 +1,491 @@
+
+### 3.3.1 Orchestration Agent
+
+### 3.3.2 Other Process
+
+## 3.4 SyncD
+
+## 3.5 SAI
+
+## 3.6 User Interface
+
+### 3.6.1 Data Models
+
+YANG models needed for NTP handling in the management framework:
+1. **openconfig-system.yang**  
+
+2. **openconfig-system-ext.yang**
+ 
+3. **sonic-system-ntp.yang**
+
+Supported yang objects and attributes:
+```diff
+
+module: openconfig-system 
+      +--rw system
+
++     +--rw ntp
++     |  +--rw config
+      |  |  +--rw enabled?                           boolean
++     |  |  +--rw ntp-source-address?                oc-inet:ip-address
+      |  |  +--rw enable-ntp-auth?                   boolean
++     |  |  +--rw oc-sys-ext:ntp-source-interface
++     |  |  |  +--rw oc-sys-ext:interface?      -> /oc-if:interfaces/interface/name
++     |  |  |  +--rw oc-sys-ext:subinterface?   -> /oc-if:interfaces/interface[oc-if:name=current()/../interface]/subinterfaces/subinterface/index
++     |  |  +--rw oc-sys-ext:vrf?                    string
+      |  +--ro state
+      |  |  +--ro enabled?                           boolean
++     |  |  +--ro ntp-source-address?                oc-inet:ip-address
+      |  |  +--ro enable-ntp-auth?                   boolean
+      |  |  +--ro auth-mismatch?                     oc-yang:counter64
++     |  |  +--ro oc-sys-ext:ntp-source-interface
++     |  |  |  +--ro oc-sys-ext:interface?      -> /oc-if:interfaces/interface/name
++     |  |  |  +--ro oc-sys-ext:subinterface?   -> /oc-if:interfaces/interface[oc-if:name=current()/../interface]/subinterfaces/subinterface/index
++     |  |  +--ro oc-sys-ext:vrf?                    string
+      |  +--rw ntp-keys
+      |  |  +--rw ntp-key* [key-id]
+      |  |     +--rw key-id    -> ../config/key-id
+      |  |     +--rw config
+      |  |     |  +--rw key-id?      uint16
+      |  |     |  +--rw key-type?    identityref
+      |  |     |  +--rw key-value?   string
+      |  |     +--ro state
+      |  |        +--ro key-id?      uint16
+      |  |        +--ro key-type?    identityref
+      |  |        +--ro key-value?   string
++     |  +--rw servers
++     |     +--rw server* [address]
++     |        +--rw address    -> ../config/address
++     |        +--rw config
++     |        |  +--rw address?            oc-inet:host
+      |        |  +--rw port?               oc-inet:port-number
+      |        |  +--rw version?            uint8
+      |        |  +--rw association-type?   enumeration
+      |        |  +--rw iburst?             boolean
+      |        |  +--rw prefer?             boolean
++     |        +--ro state
++     |           +--ro address?              oc-inet:host
+      |           +--ro port?                 oc-inet:port-number
+      |           +--ro version?              uint8
+      |           +--ro association-type?     enumeration
+      |           +--ro iburst?               boolean
+      |           +--ro prefer?               boolean
++     |           +--ro stratum?              uint8

```

Comment:
How are you getting all this stuff to the Mgmt container?

Reply:
Mgmt framework gets the corresponding peer status via ntpq utility command "ntpq -p" and translate them to the openconfig system NTP YANG. Is this what you are asking?